File: src/test/java/TestSubstr.java
import org.junit.Assert;
import org.junit.Test;


public class TestSubstr {

    @Test
    public void testSimple() {
        var fraction = TestUtils.createFraction(3, 4);
        var result = fraction.minus(new NotImplementedFraction(2, 5));

        Assert.assertEquals((Integer) 7, result.getNumerator());
        Assert.assertEquals((Integer) 20, result.getDenominator());
    }

    @Test
    public void testNormalisation() {
        var fraction = TestUtils.createFraction(3, 2);
        var result = fraction.minus(new NotImplementedFraction(1, 2));

        Assert.assertEquals((Integer) 1, result.getNumerator());
        Assert.assertEquals((Integer) 1, result.getDenominator());
    }
}

File: src/test/kotlin/TestSubstr.kt
import org.junit.Assert
import org.junit.Test


class TestSubstr {

    @Test
    fun testSimple() {
        val fraction = TestUtils.createFraction(3, 4)
        val result = fraction.minus(NotImplementedFraction(2, 5))

        Assert.assertEquals(7, result.numerator)
        Assert.assertEquals(20, result.denominator)
    }

    @Test
    fun testNormalisation() {
        val fraction = TestUtils.createFraction(3, 2)
        val result = fraction.minus(NotImplementedFraction(1, 2))

        Assert.assertEquals(1, result.numerator)
        Assert.assertEquals(1, result.denominator)
    }
}
